lib/nexmo_developer/app/presenters/head.rb

```diff
@@ -1,66 +1,96 @@
 class Head
-  attr_reader :items
+  class Description
+    def initialize(config:, frontmatter:)
+      @config      = config
+      @frontmatter = frontmatter
+    end
 
-  def initialize(items: nil)
-    @items = items
+    def description
+      @description ||= from_frontmatter || @config.fetch('description') do
+        raise "You must provide a 'description' parameter in header_meta.yml"
+      end
+    end
+
+    def from_frontmatter
+      @frontmatter && (@frontmatter['meta_description'] || @frontmatter['description'])
+    end
+  end
+
+  attr_reader :config, :frontmatter
+
+  def initialize(frontmatter = nil)
+    @frontmatter = frontmatter
 
     after_initialize!
+    validate_files_presence!
   end
 
-  def head_from_config(config)
-    config = YAML.safe_load(open_config(config))
-    {
-      title: title(config),
-      description: description(config),
-      google_site_verification: config['google-site-verification'] || '',
-      application_name: config['application-name'],
-      og_image: file(config['og-image']),
-      og_image_width: config['og-image-width'],
-      og_image_height: config['og-image-height'],
-      apple_touch_icon: file('apple-touch-icon.png'),
-      favicon: file('favicon.ico'),
-      favicon_32_squared: file('favicon-32x32.png'),
-      manifest: file('manifest.json'),
-      safari_pinned_tab: file('safari-pinned-tab.svg'),
-      mstile_144_squared: file('mstile-144x144.png'),
-    }
+  def title
+    @title ||= config.fetch('title') do
+      raise "You must provide a 'title' parameter in header_meta.yml"
+    end
   end
 
-  def open_config(config)
-    File.open(config)
+  def description
+    @description ||= Description.new(config: config, frontmatter: frontmatter).description
```

Comment:
This means the `description` parameter must be present, right? I tried with only `meta_description` in the frontmatter and the app raised an exception:

```
ActionView::Template::Error (You must provide a 'description' parameter in header_meta.yml):
     9:   <% end %>
    10:   <title><%= page_title %></title>
    11: 
    12:   <% if head.description %>
    13:     <meta name="description" content="<%= head.description %>">
    14:   <% end %>
    15: 
```

Reply:
yes, it's the same behavior as before
https://github.com/Nexmo/station/blob/master/lib/nexmo_developer/app/presenters/head.rb#L55-L59